server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -400,7 +400,7 @@ private void snapshots(String repositoryName, Collection<SnapshotId> snapshotIds
             if (cancellableTask.notifyIfCancelled(listener)) {
                 return;
             }
-            final Set<SnapshotInfo> snapshotSet = new HashSet<>();
+            final List<SnapshotInfo> snapshots = new ArrayList<>(snapshotIds.size());
```

Comment:
>No need to create a set (the values are distinct anyway)

`snapshotIds` is defined as collection in the method signature

https://github.com/elastic/elasticsearch/blob/b95cb8c40dc701cbce18308cdcd5538fd60facb9/server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java#L399

that is only called from 

https://github.com/elastic/elasticsearch/blob/b95cb8c40dc701cbce18308cdcd5538fd60facb9/server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java#L378

where `toResolve`

https://github.com/elastic/elasticsearch/blob/b95cb8c40dc701cbce18308cdcd5538fd60facb9/server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java#L328

is defined as a HashSet. I think it is worth to update the method signature to accept `Set<SnapshotId> snapshotIds` to better reflect that there are no duplicates expected.

Reply:
I know what you mean, but (a) nothing particularly bad happens if there _are_ duplicates here, and (b) we'd have to change that `.toList()` into a `collect(Collectors.toSet())` which is quite a bit more expensive. This whole thing will go away eventually.